src/dashboard/get_chart_data/templates/filter_numeric.sql.jinja

```diff
@@ -0,0 +1,48 @@
+{%- import 'syntax.sql.jinja' as syntax -%}
+
+{%- set ns =namespace(matches=[]) -%}
+{%- set filters=['eq', 'ne', 'gt', 'gte', 'lt', 'lte'] -%}
+
+{%- macro render_filter( data, filter_type, bound) -%}
+{%- if filter_type == 'eq'-%}
+{{ data }} = {{ bound }}
+{%- elif filter_type == 'ne'-%}
+{{ data }} != {{ bound }}
+{%- elif filter_type == 'gt'-%}
+{{ data }} > {{ bound }}
+{%- elif filter_type == 'gte'-%}
+{{ data }} >= {{ bound }}
+{%- elif filter_type == 'lt'-%}
+{{ data }} < {{ bound }}
+{%- elif filter_type == 'lte'-%}
+{{ data }} <= {{ bound }}
+{%- else -%}
+    not found {{ filter_type }}
+{%- endif -%}
+{%- endmacro -%}
+
+{%- macro is_filter_in_list(filter_configs) -%}
+    {%- for f in filter_configs -%}
+        {%- if f.filter_type in filters -%}
+            {%- set ns.matches = ns.matches + [f] -%}
+        {%- endif -%}
+    {% endfor -%}
+{%- endmacro -%}
+
+{%- macro get_filters(filter_configs) -%}
+    {{ is_filter_in_list(filter_configs) }}
```

Comment:
Jinja is a cursed language. I understand why you did the namespace games here. But maybe throw some comments around the place explaining that jinja does not have the ability to do actual pure-logic methods / modify variables out of scope.

But also... couldn't you just do this filtering on the python side? And then avoid all this in jinja?

Reply:
yeah, commenting will come at the end.

i :could: do the splitting on the python side - i'm then maintaining the list of filters in both jinja and python, which i don't love, but it would avoid this whole thing.

Reply:
ok i ended up doing some rework to move more of this logic into python.